backstop/src/emissions/manager.rs

```
@@ -83,70 +146,86 @@ pub fn gulp_emissions(e: &Env) -> i128 {
+
+        // distribute pool emissions via allowance to pools
+        let blnd_token_client = TokenClient::new(e, &storage::get_blnd_token(e));
+        let current_allowance = blnd_token_client.allowance(&e.current_contract_address(), pool);
+        let new_seq = e.ledger().sequence() + storage::LEDGER_BUMP_USER; // ~120 days
+        blnd_token_client.approve(
+            &e.current_contract_address(),
+            pool,
+            &(current_allowance + new_pool_emissions),
+            &new_seq,
+        );
+        set_backstop_emission_eps(e, &pool, &pool_balance, new_backstop_emissions);
+        return (new_backstop_emissions, new_pool_emissions);
     }
-    new_emissions
+    return (0, 0);
 }
 
-/// Consume pool emissions approve them to be transferred by the pool
-pub fn gulp_pool_emissions(e: &Env, pool_id: &Address) -> i128 {
-    let pool_emissions = storage::get_pool_emissions(e, pool_id);
-    if pool_emissions == 0 {
-        panic_with_error!(e, BackstopError::BadRequest);
+pub fn update_rz_emis_data(e: &Env, pool: &Address, to_gulp: bool) -> i128 {
+    if let Some(emission_data) = storage::get_rz_emis_data(e, pool) {
+        let pool_balance = storage::get_pool_balance(e, pool);
+        let gulp_index = storage::get_rz_emission_index(e);
+        let mut accrued = emission_data.accrued;
+        if emission_data.index < gulp_index || to_gulp {
+            if pool_balance.non_queued_tokens() > 0 {
+                let new_emissions = pool_balance
+                    .non_queued_tokens()
+                    .fixed_mul_floor(gulp_index - emission_data.index, SCALAR_14)
+                    .unwrap_optimized();
+                accrued += new_emissions;
+                return set_rz_emissions(e, pool, gulp_index, accrued, to_gulp);
+            } else {
+                return set_rz_emissions(e, pool, gulp_index, accrued, to_gulp);
+            }
+        }
     }
+    return 0;
```

this should panic instead of return zero.

- [ ] Either way, can we double check that `gulp_emissions` on the pool can handle the case where it gets 0 tokens to emit?

---

looks like `gulp_emissions` will panic if less than 1 BLND is emitted, so this looks OK.